channels/applicationinsights-channel-js/src/EnvelopeCreator.ts

```diff
@@ -183,6 +191,7 @@ export class DependencyEnvelopeCreator extends EnvelopeCreator {
         const customMeasurements = telemetryItem[strBaseData].measurements || {};
         const customProperties = telemetryItem[strBaseData][strProperties] || {};
         EnvelopeCreator.extractPropsAndMeasurements(telemetryItem.data, customProperties, customMeasurements);
+        EnvelopeCreator.convertPropsUndefinedToBlank(customProperties);
```

Comment:
This needs to be behind a configuration as this will affect the performance and not everyone will want this behavior. It might even be worth having the config define the "value" used for undefined.